src/devices/ServoMotor/README.md

```diff
@@ -0,0 +1,69 @@
+﻿# Servo Motor
+
+## Summary
+This is a generic binding to control many servo motors using a hardware or software `PwmChannel`.  Servo motors are usually based on a signal frequency of 50Hz.  They also require a minimum/maximum pulse width to determine the position.  The pulse width is generally between 1 and 2 milliseconds, where 1ms is approximately 0 degrees, 1.5ms is the rest position, and 2ms is 180 degrees.  This information can be found in each servo motor's datasheet.  
+
+One thing to be aware of is the wiring as the servo motor connector is usually a 3-pin connector.  The pinout is shown below where colors can vary.
+
+| Pin Number | Signal         | Color                     |
+|------------|----------------|---------------------------|
+| 1          | Ground         | Black or Brown            |
+| 2          | Power Supply   | Brown or Red              |
+| 3          | Control Signal | Orange or White or Yellow |
+
+
+## Device Family
+There are many servo motor sizes available that offer both standard and continuous rotation.  Below are a few links where to purchase servo motors.
+
+[Adafruit Servo Motor Accessories](https://www.adafruit.com/?q=servo)  
+[Sparkfun Servo Motor Accessories](https://www.sparkfun.com/categories/245)  
+
+## Binding Notes
+
+The `ServoMotor` binding offers an easy way to begin controlling a servo motor.  The quickest approach is to provide the `ServoMotor` object a `PwmChannel` using the default values for other optional arguments.
+
+```csharp
+// Example of hardware PWM using chip 0 and channel 0 on a dev board.
+ServoMotor servoMotor = new ServoMotor(PwmChannel.Create(0, 0, 50));
+servoMotor.Start();  // Enable control signal.
+
+// Move position.  Pulse width argument is in microseconds.
+servoMotor.WritePulseWidth(1000); // 1ms; Approximately 0 degrees.
+servoMotor.WritePulseWidth(1500); // 1.5ms; Approximately 90 degrees.
+servoMotor.WritePulseWidth(2000); // 2ms; Approximately 180 degrees.
+
+servoMotor.Stop(); // Disable control signal.
+```
+
+The position of servo motor can also be adjusted by the angle.  The `ServoMotor` constructor's optional arguments must be set according to device's specs.  NOTE: These are usually an approximation, so you may need to manually tweak to determine exact values.
+
+For example, the [Hitec HS-300BB](https://servodatabase.com/servo/hitec/hs-300bb) servo has the following specifications:
+- MaximumAngle = 180
+- MinimumPulseWidthMicroseconds = 900
```

Comment:
How can pulse width length be less than PWM period (`1/frequency=1/50=200ms`)? I find this name rather confusing

Reply:
> How can pulse width length be less than PWM period

I'm not sure I understand question.  The PWM period is the entire time it takes for one cycle of frequency.  The pulse width is the duration the pulse is active.  As in your equation, servo frequencies are usually 50Hz so the period is 20ms.

The min/max widths represents the pulse width needed to position.  Properties are in microseconds, where min is 900uS (or 0.9mS) and min is 2100uS (or 2.1mS).  

![image](https://user-images.githubusercontent.com/2956825/62047779-467b0e00-b1d9-11e9-9a31-4ab4931a30da.png)

![image](https://user-images.githubusercontent.com/2956825/62047791-4aa72b80-b1d9-11e9-9de1-bf49f20fc929.png)

Reply:
LOL, seems I had a bad idea on how PWM works - I thought the frequency was relating to granularity of the fill and 50% with 50Hz frequency means sending 10101... over the wire and swap value every 20ms but that doesn't seem to be the case. Now I'm wondering if we should have an API telling us on how accurate can we be with pulse width - so basically clock frequency or whatever it is called.

Reply:
> we should have an API telling us on how accurate can we be with pulse width

That is basically what PWM is all about.  The frequency states how many times a cycle can occur in a second and the duty cycle states how long the pulse is active during each cycle.

So if you have a 1Hz signal (1 cycle a second) and a duty cycle of 50%, the pulse width would be 0.5s/500ms/500000us.

In this Servo Motor binding update, WritePulseWidth is setting how accurate the pulse should be active.  You could technically move this method to PwmChannel classes (and I started to), but thought it would be a longer review/discussion on having it added to core API.

😄 sometimes i hate technology

Reply:
yep, unfortunately issue first for System.Device.Gpio at least :smile: